### PR TITLE
Add Missing Entity API: Fox

### PR DESCRIPTION
**Pull request by Owen1212055**
This adds a new patch that can incorporate missing entity API. This first one starts of with fox api.

Certain behaviors have additional animations as well.
So cute!!!
image
Fox#setHeadTilted

image
Fox#setLeaping

### CODE REVIEW

**Review by electronicboy — APPROVED** [review]

**Comment by Machine-Maker** [comment]
You seem to have not committed your OBFHELPERs cause some of those methods don't exist on EntityFox

**Comment by Owen1212055** [comment]
> You seem to have not committed your OBFHELPERs cause some of those methods don't exist on EntityFox

Ahhh... good catch there.

**Comment by Owen1212055** [comment]
Should be good now

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+diff --git a/src/main/java/net/minecraft/world/entity/animal/EntityFox.java b/src/main/java/net/minecraft/world/entity/animal/EntityFox.java
+index b7fa24318ef43918b6b10ff4ea8acb960527296e..c1c5c8fdc603877b39506a319f90294bbf840c2c 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/EntityFox.java
++++ b/src/main/java/net/minecraft/world/entity/animal/EntityFox.java
+@@ -397,18 +397,22 @@ public class EntityFox extends EntityAnimal {
+         this.d(1, flag);
+     }
+ 
++    public boolean isWalking() {return eN();} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public boolean isFaceplanted() {return eN();} // Paper - OBFHELPER
```

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+     public boolean eN() {
+         return this.t(64);
+     }
+ 
++    public void setWalking(boolean walking) {x(walking);} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public void setFaceplanted(boolean faceplanted) {x(faceplanted);} // Paper - OBFHELPER
```

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+     private void x(boolean flag) {
+         this.d(64, flag);
+     }
+ 
++    public boolean isAngry() {return fb();} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public boolean isDefending() {return fb();} // Paper - OBFHELPER
```

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+     private boolean fb() {
+         return this.t(128);
+     }
+ 
++    public void setAngry(boolean angry) {y(angry);} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public void setDefending(boolean defending) {y(defending);} // Paper - OBFHELPER
```

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+ 
++    public void setLeaping(boolean leaping) {u(leaping);} // Paper - OBFHELPER
+     public void u(boolean flag) {
+         this.d(16, flag);
+     }
+@@ -561,10 +567,12 @@ public class EntityFox extends EntityAnimal {
+         return this.t(4);
+     }
+ 
++    public void setHeadTilted(boolean tilted) {w(tilted);} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public void setInterested(boolean interested) {w(interested);} // Paper - OBFHELPER
```

File: Spigot-Server-Patches/0739-Fox-API.patch [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+     public void w(boolean flag) {
+         this.d(8, flag);
+     }
+ 
++    public boolean isHeadTilted() {return eW();} // Paper - OBFHELPER
```
Comment (kennytv) [inline Spigot-Server-Patches/0739-Fox-API.patch]:
```suggestion
+    public boolean isInterested() {return eW();} // Paper - OBFHELPER
```

**Review by kennytv — CHANGES_REQUESTED** [review]
... and so on

**Comment by Owen1212055** [comment]
Combined: See https://github.com/PaperMC/Paper/pull/5744#issuecomment-850769784

**Comment by Owen1212055** [comment]
Updated to 1.17

**Review by kennytv — APPROVED** [review]
Thank you

**Pull request merged**